### DISCUSSION TITLE
R - Error saat menjalankan workflow_map()

### DISCUSSION THREAD

**Discussion by Hertzspr** [teamalgoritma/community#266] [Q&A] [answered]
I was trying to use workflow_set() from tidymodels approach to run several model simultaneusly. 

``
not_normalized <- 
   workflow_set(
      preproc = list(rec = recipe), 
      models = list(LOG_reg = lr_mod, 
                    NAIVE_bay = nb_mod,
                    RANDOM_f = rf_mod)
   )

normalized <- 
   workflow_set(
      preproc = list(normalized = norm_recipe), 
      models = list(SVM_poly = svmp_mod, 
                    KNN = knn_mod)
   )

all_workflow <-
  bind_rows(not_normalized, normalized)

fit_res <- 
  all_workflow %>% 
  workflow_map(
    verbose = T,
    seed = 1,
    resamples = val_set,
    control = control_grid(save_pred = T)
  )
``

Here's the error: 

Error in pkgs$pkg[[1]] : subscript out of bounds
8.
get_pkgs(x, infra)
7.
required_pkgs.model_spec(model, infra = infra)
6.
generics::required_pkgs(model, infra = infra)
5.
required_pkgs.workflow(w, infra = FALSE)
4.
generics::required_pkgs(w, infra = FALSE)
3.
has_all_pkgs(wflow)
2.
workflow_map(., verbose = T, seed = 1, resamples = val_set, control = control_grid(save_pred = T))
1.
all_workflow %>% workflow_map(verbose = T, seed = 1, resamples = val_set, 
control = control_grid(save_pred = T))

---

Any idea about what's the problem? I followed tutorials from tmwr.org when I was doing this. But I didn't follow it exactly. I didn't prepare the models for tuning with tune(), since I don't want to do any hypertuning, I just want to see how each models performed with their default parameters. 

---

Never mind, the previous error was caused by my fault of not installing and loading the required library. But now I have another problem: 

Error in `halt()`:
! There were 1 workflows that had no results.

the one that didn't work is naive_bayes. 

``nb_mod <-
  naive_Bayes() %>% 
  set_engine("klaR") %>% 
  set_mode("classification")``

-

Nevermind. I forgot to install klaR. So this is wrapped up. It works. I want to delete this post but I see no delete button.

**Comment by amaliasf** [comment, marked as answer]
Dear @Hertzspr 

Kami senang Bapak sudah dapat menyelesaikan problem tersebut mandiri. Terima kasih telah menginfokan juga cara solvingnya. Kami izin menutup diskusi ini.
